src/Microsoft.PowerShell.Commands.Utility/commands/utility/GetDateCommand.cs

```diff
@@ -212,26 +235,73 @@ public int Millisecond
         /// <summary>
         /// Unix format string.
         /// </summary>
-        [Parameter(ParameterSetName = "UFormat")]
+        [Parameter]
         [ValidateNotNullOrEmpty]
-        public string UFormat { get; set; }
+        public string UFormat
+        {
+            get
+            {
+                return _uFormat;
+            }
+
+            set
+            {
+                _uFormat = value;
+                _uFormatSpecified = true;
+            }
+        }
+
+        private string _uFormat;
+        private bool _uFormatSpecified;
 
         /// <summary>
-        /// Unix format string.
+        /// .NET format string.
         /// </summary>
-        [Parameter(ParameterSetName = "net")]
+        [Parameter]
         [ArgumentCompletions("FileDate", "FileDateUniversal", "FileDateTime", "FileDateTimeUniversal")]
-        public string Format { get; set; }
+        public string Format
+        {
+            get
+            {
+                return _format;
+            }
+
+            set
+            {
+                _format = value;
+                _formatSpecified = true;
```

Comment:
We have no need to add the field - use `this.MyInvocation.BoundParameters[]` to check if a parameter is present.

Reply:
Perhaps we could create parameter sets to resolve this.

Reply:
For Format and UFormat, I made it parameter set (thanks to @mklement0).

For Date and UnixTimeSeconds, the other parameters are not checked by BoundParameters, and use a discernment field, so I'm following them.